wordpress-mobile/wellsql
# [Build] Update Gradle to 7.4.2 & AGP to 7.2.1

This PR upgrades Gradle to [7.4.2](https://docs.gradle.org/7.4.2/release-notes.html) and AGP to [7.2.1](https://developer.android.com/build/releases/past-releases/agp-7-2-0-release-notes).

FYI: This PR also relates to this FluxC [#2766](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2766) PR and the fact that the `com.android.support` group dependency had to be excluded from the `wellsql` dependency to get the project to build without `AndroidX` and `android.enableJetifier=true` related warnings (see PR and [commit](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2766/commits/251b1a5cdbf4caf76171b529db5c79bb67c70b50) description).

PS: In addition to this Gradle and AGP upgrade, the below changes were added in order to align this repo with the rest of such [WordPress-Mobile](https://github.com/wordpress-mobile) Android library related repos.

-----

Build List:

1. [Disable the android enable jetifier gradle property](https://github.com/wordpress-mobile/wellsql/commit/817189003ddff43c6d9ffefb9d1a7f4c7b77a0a9)
2. [Replace com android support with androidx annotation dependency](https://github.com/wordpress-mobile/wellsql/commit/1a58f9fc4091169aea0bcfd649817226042e080c)
3. [Replace com android support test with androidx test dependency](https://github.com/wordpress-mobile/wellsql/commit/bee1ba95fe475034008a82838222113f6cc044a7)
4. [Enable non transitive resources for the project](https://github.com/wordpress-mobile/wellsql/commit/677c87fe585c003edaaf45edc47c6735f4954dd4)
5. [Remove empty gradle properties file from well processor module](https://github.com/wordpress-mobile/wellsql/pull/24/commits/283b9053d0927a1ced80cd67fe0a9b06924664fe)
6. [Downgrade androidx annotation version to 1.2.0](https://github.com/wordpress-mobile/wellsql/pull/24/commits/1b57761ba47a7bbc6233976f2adcfbf8016729ef)
7. [Add copy gradle properties command to all publishing steps](https://github.com/wordpress-mobile/wellsql/pull/24/commits/c034211e218abd5134c6aa754f96f8f9bfec7c92)

Lint List:

1. [Suppress expired target sdk version for all modules](https://github.com/wordpress-mobile/wellsql/commit/52810544b56a4de253bffb932be36f61944a047e)
2. [Add lint check to buildkite](https://github.com/wordpress-mobile/wellsql/commit/f29c6153e08556a907a8448d6a183b21e166ff3a)

Test Restructure List:

1. [Update junit to 4.13.2](https://github.com/wordpress-mobile/wellsql/commit/1dd3d9dddd474e2cce1380c2da07af9bcda9b130)
2. [Remove all unit test related configuration from project](https://github.com/wordpress-mobile/wellsql/commit/84548f89a5214eac9dc8484017205c736fa53adc)

Dependency Versions Refactor List:

1. [Extract all sdk version to a separate ext block](https://github.com/wordpress-mobile/wellsql/commit/92ecfe26274af2544dd997e4b2b0e0c71d6588da)
3. [Extract google auto service version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/f746b72b044e63705f13d1e8bf438666c1986957)
4. [Extract squareup java poet version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/5513c94fdb570fffb51a76ee34a286c4792ca4b4)
6. [Extract androidx annotation version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/aea9159a2105f03418b2f3b01ecb3918465291f0)
7. [Extract junit version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/aa6fa708cf3e162c494de0d40ab17f7e12b19496)
8. [Extract androidx appcompat version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/f900c8d24a8689aa24794897700c674be55c186d)
9. [Extract androidx test core version to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/987e7dd601d3255f736e0b4fbab28d2c26721f8f)
10. [Add versions groups to root build gradle](https://github.com/wordpress-mobile/wellsql/commit/fbadeb5756a56e963a8c23bcbb90559d2a711ea4)
11. [Extract automattic publish to s3 version to settings build gradle](https://github.com/wordpress-mobile/wellsql/pull/24/commits/aca425ba2335b991e4d0876d1169b2e38e83cb72)

-----

## To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could run `./gradlew build` locally and verify that  it is successful (see [previous](https://github.com/wordpress-mobile/wellsql/pull/19) such update).

-----

I also noticed the below AGP `8.0` warning, which relates to the new publishing DSL. This is maybe something we need to look into for this and all other such library related repos that utilize such publishing functionality. Wdyt @oguzkocer ?

```
> Configure project :wellsql
WARNING:Software Components will not be created automatically for Maven publishing
from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property
android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use
the new publishing DSL.
```
